### PR TITLE
Fix slurm resource assignment issue (#495)

### PR DESCRIPTION
**Pull request by lgannoaa**
The issue:
Some jobs are running on Hera using old job cards. It does not match the requested tasks with the number of requested nodes. This caused the slurm task scheduler to issue a Warning message and ignored the setting.
During a test using global-workflow develop on Hera the following issues were found. (#495)

gdasanaldiag.log (3936)
7.794 + srun -l '--export=ALL' -n 90 --multi-prog /scratch1/NCEPDEV/stmp2/Lin.Gan/RUNDIRS/naf19296/2020090200/gdas/anal.311060/mp_diag.sh
srun: Warning: can't honor --ntasks-per-node set to 40 which doesn't match the requested tasks 90 with the number of requested nodes 3. Ignoring --ntasks-per-node.
gdasanalcalc.log (2181)
0.075 + srun -l '--export=ALL' -n 1 /scratch1/NCEPDEV/global/Lin.Gan/git/gw-launcher-flag/exec/gaussian_sfcanl.exe
srun: Warning: can't run 1 processes on 4 nodes, setting nnodes to 1
gdasanal.log (7130, 8711)
0 + srun -l --export=ALL -n 6 /scratch1/NCEPDEV/global/Lin.Gan/git/gw-launcher-flag/exec/global_cycle '1>OUTPUT.111018' '2>errfile'
srun: Warning: can't run 6 processes on 11 nodes, setting nnodes to 6
gdaseupd.log (2927)
2.589 + srun -l '--export=ALL' -n 1 --multi-prog /scratch1/NCEPDEV/stmp2/Lin.Gan/RUNDIRS/naf19296/2020090200/gdas/eupd.305717/mp_untar.sh
srun: Warning: can't run 1 processes on 40 nodes, setting nnodes to 1
gdaspost001.log (3961,10545)
25.797 + srun -l '--export=ALL' -n 24 --multi-prog /scratch1/NCEPDEV/stmp2/Lin.Gan/RUNDIRS/naf19296/2020090200/gdas/post.212378/poescript_srun
srun: Warning: can't honor --ntasks-per-node set to 12 which doesn't match the requested tasks 24 with the number of requested nodes 10. Ignoring --ntasks-per-node.
gfsanal.log (6668)
0 + srun -l --export=ALL -n 6 /scratch1/NCEPDEV/global/Lin.Gan/git/gw-launcher-flag/exec/global_cycle '1>OUTPUT.94444' '2>errfile'
srun: Warning: can't run 6 processes on 11 nodes, setting nnodes to 6
gfsanalcalc.log (1735)
0.067 + srun -l '--export=ALL' -n 1 /scratch1/NCEPDEV/global/Lin.Gan/git/gw-launcher-flag/exec/gaussian_sfcanl.exe
srun: Warning: can't run 1 processes on 4 nodes, setting nnodes to 1
gfspost001.log (3970,10554,18492,25264,33210,39982,47928,54700)
4.319 + srun -l '--export=ALL' -n 24 --multi-prog /scratch1/NCEPDEV/stmp2/Lin.Gan/RUNDIRS/naf19296/2020090200/gfs/post.40235/poescript_srun
srun: Warning: can't honor --ntasks-per-node set to 12 which doesn't match the requested tasks 24 with the number of requested nodes 10. Ignoring --ntasks-per-node.

Technical Updates:
  Modified:   env/HERA.env and env/ORION.env
    unset SLURM_NTASKS_PER_NODE
      (to fix gdasanaldiag issue)
    Modify resource assignment for APRUN_CYCLE and APRUN_GAUSFCANL with analysis job
      (to fix gdas/gfs gdasanalcalc gfsanalcalc gaussian_sfcanl.exe resource assignment issue)
      (to fix gdas/gfs analysis global_cycle resource assignment issue)
    Modify resource assignment for APRUNCFP in eupd job
      (to fix eupd untar script resource assignment issue)
  Modified:   parm/config/config.post
    Move npe_dwn to config.resource
      (To keep resource setting under config.resource)
  Modified:   parm/config/config.resources
    Add logic to eval if running in-line post for correct post resource assignment
    Fix the resource assignment issue for post job
    Fix incorrect wall time limit for post job

### CODE REVIEW

File: env/HERA.env [hunk 1]
```diff
@@ -72,12 +72,12 @@ elif [ $step = "anal" ]; then
     export NTHREADS_CYCLE=${nth_cycle:-12}
     [[ $NTHREADS_CYCLE -gt $npe_node_max ]] && export NTHREADS_CYCLE=$npe_node_max
     npe_cycle=${ntiles:-6}
-    export APRUN_CYCLE="$launcher -n $npe_cycle"
+    export APRUN_CYCLE="$launcher -N 1 -n $npe_cycle"
```
Comment (aerorahul) [inline env/HERA.env:75]: Is hardwiring `-N 1` the way to go?  What happens if one wishes to spread it across multiple nodes?

File: parm/config/config.resources [hunk 2]
```diff
@@ -195,6 +195,16 @@ elif [ $step = "post" ]; then
...
     export npe_node_dwn=$npe_node_max
+    if [ $machine = "WCOSS_DELL_P3" ]; then
+      export npe_dwn=28
+    else
+      export npe_dwn=24
+    fi
+    if [ "$WRITE_DOPOST" = ".true." ]; then 
+      export npe_post=$npe_dwn
+      export wtime_post="00:10:00"
+      export wtime_post_gfs="00:10:00"
```
Comment (aerorahul) [inline parm/config/config.resources:206]: How does this compare with the lines 192, 193

File: parm/config/config.resources [hunk 3]
```diff
@@ -195,6 +195,16 @@ elif [ $step = "post" ]; then
     export nth_post=1
     export npe_node_post=12
     export npe_node_dwn=$npe_node_max
+    if [ $machine = "WCOSS_DELL_P3" ]; then
+      export npe_dwn=28
+    else
+      export npe_dwn=24
+    fi
+    if [ "$WRITE_DOPOST" = ".true." ]; then 
+      export npe_post=$npe_dwn
```
Comment (aerorahul) [inline parm/config/config.resources:204]: This is significantly different than the one on line 194.
Can you explain in short what is `WRITE_DOPOST` intended use?

**Review by aerorahul — CHANGES_REQUESTED** [review]
See comments.

File: env/HERA.env (hunk 1 above)
Comment (lgannoaa) [reply env/HERA.env:75]: The current design is only use N 1 configuration. Multiple change will need to be done if across multiple nodes is become a requirement. The code manager can make that change as need. The N 1 solution is representing the fact of this job in the current state.

File: parm/config/config.resources (hunk 2 above)
Comment (lgannoaa) [reply parm/config/config.resources:206]: lines 192, 193 is not reality when running with inline-post. The job only need this amount of time when inline-post is turned on.

File: parm/config/config.resources (hunk 3 above)
Comment (lgannoaa) [reply parm/config/config.resources:204]: The WRITE_DOPOST is a switch for inline-post. When it is turned on, the post job use much less system resource. This modification will allow post job to use less resource and get to run faster in supercomputer.

File: parm/config/config.resources (hunk 2 above)
Comment (lgannoaa) [reply parm/config/config.resources:206]: Answered in File Changed page.

File: parm/config/config.resources (hunk 3 above)
Comment (lgannoaa) [reply parm/config/config.resources:204]: Answered in File Changed page.

**Comment by KateFriedman-NOAA** [comment]
This seems to be languishing...what is the status of this? Can this be closed and reevaluated?

**Comment by lgannoaa** [comment]
Is the solution provided been merged?

**Comment by KateFriedman-NOAA** [comment]
Still not clear what issue this addresses and no solution has been made available. Closing and will be reevaluated if needed down the road.

**Pull request closed without merging**